pgoapi/pgoapi.py

```diff
@@ -623,39 +639,65 @@ def cleanup_pokemon(self, inventory_items=None):
+                                                                              pokemons_keep_counter,
+                                                                              pokemons_keep_iv_counter)
+                    if (eligible and pokemons_keep_counter > self.MIN_SIMILAR_POKEMON)\
+                            or pokemons_keep_counter >= self.MAX_SIMILAR_POKEMON:
                         self.do_release_pokemon(pokemon)
+                        continue
+                    else:
+                        pokemons_keep_counter += 1
+                        if keep_iv:
+                            pokemons_keep_iv_counter += 1
 
-    def is_pokemon_eligible_for_transfer(self, pokemon, best_pokemon):
+    def is_pokemon_eligible_for_transfer(self, pokemon, best_pokemon=Pokemon(),
+                                         pokemons_keep_counter=0, pokemons_keep_iv_counter=0):
         # never release favorites and other defined pokemons
         if pokemon.is_favorite or pokemon.pokemon_id in self.keep_pokemon_ids:
-            return False
-        elif self.RELEASE_DUPLICATES and (
-                    self.pokemon_lvl(best_pokemon) * self.RELEASE_DUPLICATES_SCALER > self.pokemon_lvl(
-                    pokemon) and self.pokemon_lvl(pokemon) < self.RELEASE_DUPLICATES_MAX_LV):
-            return True
-        # release defined throwaway pokemons  but make sure we have kept at least 1 (dont throw away all of them)
-        elif pokemon.pokemon_id in self.throw_pokemon_ids:
-            return True
+            return False, False
+        # release defined throwaway pokemons, but make sure we have kept at least 1 (dont throw away all of them)
+        elif pokemon.pokemon_id in self.throw_pokemon_ids and pokemons_keep_counter >= self.MIN_SIMILAR_POKEMON:
+            return True, False
+        # release duplicates if applicable
+        elif self.RELEASE_METHOD == "DUPLICATES":
+            if (self.pokemon_score(best_pokemon) * self.RELEASE_DUPLICATES_SCALAR > self.pokemon_score(pokemon) and
+                    self.pokemon_score(pokemon) < self.RELEASE_DUPLICATES_MAX_SCORE) and \
+                            pokemons_keep_counter >= self.MIN_SIMILAR_POKEMON:
+                return True, False
+            else:
+                return False, False
+        # keep high-iv pokemons based on config values
+        elif pokemon.iv_normalized > self.KEEP_IV_OVER \
+                and pokemons_keep_iv_counter < self.MAX_POKEMON_HIGH_IV \
+                and pokemon.cp > (best_pokemon.cp * self.KEEP_IV_WITH_PERCENT_CP / 100):
+            return False, True
         # keep high-cp pokemons
-        elif pokemon.cp > self.KEEP_CP_OVER or pokemon.iv > self.KEEP_IV_OVER:
-            return False
-        # if we haven't found a reason to keep it, transfer it
-        else:
-            return True
-
-    def pokemon_lvl(self, pokemon):
```

Comment:
good suggestion, will pass SCORE_METHOD and SCORE_SETTINGS to the pokemon class...